signal/signal.go

```diff
@@ -15,21 +15,21 @@ import (
 
 var (
 	// interruptChannel is used to receive SIGINT (Ctrl+C) signals.
-	interruptChannel = make(chan os.Signal, 1)
+	interruptChannel chan os.Signal
```

Comment:
could we instead encapsulate these in a struct?

so it could be used like
```go
signal := signal.Intercept()
<-signal.ShutdownChan()

```

Reply:
Why does this matter? In my understanding a simple encapsulation like this won't change the fact that these channels will still live as globals under `signal`. It will just require some unnecessary changes on lnd entry points and this confusing `signal := signal` assignment.

Or shouldn't the channels live as globals under `signal`? (That would require relatively big channels on that package.)

Reply:
I think it would be possible to create a single `Interceptor` struct which contains all of the channels that we use in the signal package and then define the various functions that currently reference the globals (eg `Alive`/`Listening`)  as functions on that struct? 

The downside there is that we need to pass around a single `interceptor` struct to the places where we currently call these functions, but it doesn't look too bad? The only tricky place I can see is in `DefaultConfig` because we haven't called `Intercept` yet, but we can just switch in a non-nil interceptor when we `LoadConfig` - happy to help out with this part in more detail, because I added the change which needs the interceptor for our logger!

Reply:
Yeah, my point was to remove them from the global scope entirely :)

Reply:
+1 for moving to contained struct. this will also be useful in the future if we want to run multiple lnd's in the same process, as otherwise they would all share the same global signals.